### PR TITLE
homePage and listProducts and Slider component

### PR DESCRIPTION
**Pull request by namiron**

### CODE REVIEW

File: src/common/constants.js [hunk 1]
```diff
@@ -27,3 +27,25 @@ export const ACCESSORIES = 'Accessories'
 
 //-----------------------------------------------header
 
+
+//---------------------------------------pagination
+
+export const itemsPerPage = 16;
+
```
Comment (Tamir198) [inline src/common/constants.js:34]: For constants use `ITEMS_PER_PAGE` convention

File: src/components/ProductList/ProductItem.1.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import Image from '../../construction/Image';
+import { AddToCart, FavoriteBtn } from './../../common/Buttons';
+import global from '../../modules/global.module.scss';
+import product from '../../modules/products.module.scss';
+
```
Comment (Tamir198) [inline src/components/ProductList/ProductItem.1.jsx]: Use absolute imports not relative ones

File: src/components/ProductList/ProductItem.1.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export const ProductItem = ({ image, title, price }) => {
+
+	return (
+		<div className={product.itemBody}>
+			<div className={product.boxImage}>
+				<Link to='/item_details'>
+					<div className={product.curtain}>
+						<div className={product.box}>
+							<FavoriteBtn />
```
Comment (Tamir198) [inline src/components/ProductList/ProductItem.1.jsx]: Use destructuring on product as well

File: src/components/ProductList/ProductList.jsx [hunk 4]
```diff
@@ -1,7 +1,72 @@
...
-	return
-}
+
+	useExportAllProducts()
+	const [isLoading, productList, error] = useCustomProductList()
+
+	//----------------------------------------------------------const
+	const totalItems = productList.length
+	const [currentPage, setCurrentPage] = useState(1)
+	const totalPages = Math.ceil(totalItems / itemsPerPage)
```
Comment (Tamir198) [inline src/components/ProductList/ProductList.jsx]: Don't use magic numbers (some random number without explanation), save this value into a variable or constants

In this example its clear but take this as a thumb rule generally

File: src/components/ProductList/ProductList.jsx [hunk 5]
```diff
@@ -1,7 +1,72 @@
...
+
+	return (
+		<>
+			<div className={product_list.productListBody}>
+				<div className={product_list.productListHolder}>
+					{displayItemsOnPage()}
+				</div>
+				<div className={product_list.paginationRow}>
+					<button onClick={goToPreviousPage} className={product_list.paginationArrowPrevios} disabled={currentPage === 1}>
+						{'<'}
```
Comment (Tamir198) [inline src/components/ProductList/ProductList.jsx:55]: Be consistent with your naming, either : 

`productList` and `currentPage`
or 
`product_list` and _currentPage` (Better use camel case in js)

File: src/components/pages/Home.jsx [hunk 6]
```diff
@@ -1,19 +1,28 @@
...
-
   return (
     <>
-      <section className='home'>
-        Home
-
+      <section className={home.home}>
+        <div className={home.slider}>
+          <Slider />
+        </div>
```
Comment (Tamir198) [inline src/components/pages/Home.jsx:21]: Use destruction on home

File: src/construction/Slider/Slider.jsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+		initialIndex: 2,
+	};
+
+	const girls = {
+		id: [1, 2, 3],
+		image: [girlOne, girlsTwo, girlsThree]
+	}
+	return (
+		<div className={flickity.sliderBody}>
+			<div className={flickity.container}>
```
Comment (Tamir198) [inline src/construction/Slider/Slider.jsx:22]: Use destruction

File: src/construction/Slider/Slider.jsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+				<div className={flickity.SliderTextBg}><img src={logoSlider} alt="" /></div>
+				<Flickity
+					className={'Slider'}
+					elementType={'div'}
```
Comment (Tamir198) [inline src/construction/Slider/Slider.jsx:26]: why not `className="Slider"`

File: src/construction/Slider/Slider.jsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+					disableImageLoaded={false}
+					options={flickityOptions}
+					reloadOnUpdate
+					static
+				>
+					{girls['id'].map((i) => {
+						return (
```
Comment (Tamir198) [inline src/construction/Slider/Slider.jsx:33]: You don't need to access the id and map it (what if the id will not be numbers) 

You can do something like this : 

```js
girls.map((item,index) => ...))
```

File: src/construction/Slider/Slider.jsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+							<div key={i} className={flickity.slide}>
+								<img src={`${girls['image'][i - 1]}`} alt="girls" />
+							</div>
+						)
+					})}
+				</Flickity >
+				<div className={flickity.exploreBlock}>
+					<p className={flickity.explore}>Explore our collection</p>
+					<img src={iconsArrow} alt="" />
```
Comment (Tamir198) [inline src/construction/Slider/Slider.jsx:42]: Extract into constants

File: src/construction/Slider/slider.scss [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+ .flickity-button:active {
+	opacity: 0.6;
+ }
+ 
+ .flickity-button:disabled {
+	opacity: 0.3;
+	cursor: auto;
+	/* prevent disabled button from capturing pointer up event. #716 */
+	pointer-events: none;
+ }
```
Comment (Tamir198) [inline src/construction/Slider/slider.scss:72]: I would remove this comment, a developer that know CSS will know that `pointer-events: none` is

File: src/modules/home.module.scss [hunk 12]
```diff
@@ -0,0 +1,59 @@
+@import '../common/variables.scss';
+@import '../null.scss';
+
+
+//-----------------------------global
+.home {
+}
+.container {
```
Comment (Tamir198) [inline src/modules/home.module.scss:7]: Remove empty selector

File: src/modules/home.module.scss [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+	@extend %container ;
+}
+//-----------------------------global
+
+
+.slider{}
+
```
Comment (Tamir198) [inline src/modules/home.module.scss]: Same here

File: src/modules/productList.module.scss [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+	@extend %displayFlexDirection;
+	align-items: center;
+}
+.productListHolder {
+	@extend %gridColumn;
+	@include adaptive-value('margin-bottom',50,25,1440);
+}
+.paginationRow {
+	width: fit-content;
+	@extend %displayComplect;
```
Comment (Tamir198) [inline src/modules/productList.module.scss]: Shouldn't this be under `pagination` and not `global` comment?

File: src/modules/products.module.scss [hunk 15]
```diff
@@ -11,8 +11,10 @@
 
 .itemBody {
 	max-width: 296px;
+
 }
```
Comment (Tamir198) [inline src/modules/products.module.scss:15]: Add prettier to the project

File: src/components/ProductList/ProductItem.1.jsx (hunk 3 above)
Comment (namiron) [reply src/components/ProductList/ProductItem.1.jsx]: I don't undestand what? it is styles syntax

**Pull request merged**